### PR TITLE
Rifeljm/#1634 nft gallery user folders

### PR DESCRIPTION
**Pull request by rifeljm**

### CODE REVIEW

File: packages/core/src/components/UserFolders/AddUserFolderDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      aria-labelledby="alert-dialog-title"
+      aria-describedby="alert-dialog-description"
+      open={open}
+    >
+      {title && <DialogTitle id="alert-dialog-title">{title}</DialogTitle>}
+      <DialogContent id="alert-dialog-description">
+        <TextField
+          ref={inputWrapper}
+          autoFocus
+          onChange={(e) => {
```
Comment (seeden) [inline packages/core/src/components/UserFolders/AddUserFolderDialog.tsx:51]: do not use onChange

File: packages/core/src/components/UserFolders/AddUserFolderDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          label={placeholderName}
+          variant="filled"
+        />
+      </DialogContent>
+
+      <DialogActions>
+        <Button onClick={handleClose} variant="outlined" color="primary" autoFocus>
+          <Trans>Cancel</Trans>
+        </Button>
+        <Button
```
Comment (seeden) [inline packages/core/src/components/UserFolders/AddUserFolderDialog.tsx:69]: use handleSubmit and type="submit" or submit property
use Button from @chia-network/core

File: packages/gui/src/components/nfts/NFTCard.tsx [hunk 3]
```diff
@@ -90,6 +100,107 @@ function NFTCard(props: NFTCardProps) {
...
+  }
+
+  let startPositionLeft: number;
+  let startPositionTop: number;
+  let draggingInterval: any;
+  const countdownSpeed = 30; /* less is faster */
+  let countDown: number = countdownSpeed;
+
+  function adjustPositionBeforeAdjusted(e: any) {
+    (document.getElementById('draggable-card') as HTMLElement).style.left = `${
```
Comment (seeden) [inline packages/gui/src/components/nfts/NFTCard.tsx:145]: do not use document.getElementById instead use useRef and ref in renderCard

File: packages/gui/src/components/nfts/NFTCard.tsx [hunk 4]
```diff
@@ -90,6 +100,107 @@ function NFTCard(props: NFTCardProps) {
...
+        adjustPositionBeforeAdjusted(e);
+      }
+    }
+  }
+
+  return (
+    <Flex
+      flexDirection="column"
+      flexGrow={1}
+      onMouseDown={(e: MouseEvent) => {
```
Comment (seeden) [inline packages/gui/src/components/nfts/NFTCard.tsx:184]: use useCallback and create proper handle for this functionality

File: packages/gui/src/components/nfts/NFTCard.tsx [hunk 5]
```diff
@@ -90,6 +100,107 @@ function NFTCard(props: NFTCardProps) {
...
+        widthBeforeDrag.current = rect.width;
+        e.preventDefault();
+        setMouseDownState(true);
+        document.body.addEventListener('mouseup', mouseUpEvent);
+        document.body.addEventListener('mousemove', mouseMoveEvent);
+        setDraggedNFT(nftId);
+        countDown = countdownSpeed;
+      }}
+    >
+      <div ref={draggableContainerRef}>
```
Comment (seeden) [inline packages/gui/src/components/nfts/NFTCard.tsx:200]: do not use <div directly anywhere. 
use Box from material-ui

File: packages/gui/src/components/nfts/NFTContextualActions.tsx [hunk 6]
```diff
@@ -573,6 +577,55 @@ function NFTInvalidateContextualAction(props: NFTInvalidateContextualActionProps
...
+/* ========================================================================== */
+
+type NFTAddToGalleryContextualActionProps = NFTContextualActionProps;
+
+function NFTAddToGalleryContextualAction(props: NFTAddToGalleryContextualActionProps) {
+  const { selection } = props;
+
+  const openDialog = useOpenDialog();
+  const disabled = selection?.items.length === 0;
+  const [userFoldersNFTs, setUserFoldersNFTs] = useLocalStorage('user-folders-nfts', {});
```
Comment (seeden) [inline packages/gui/src/components/nfts/NFTContextualActions.tsx:591]: usePrefs instead of useLocalStorage
user can move his configuration file to another computer

File: packages/gui/src/components/nfts/NFTSaveToFilterDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  nftIds: string[];
+}
+
+export default function AppVersionWarning(props: NFTSaveToFilterDialogProps) {
+  const [open, setOpen] = React.useState<boolean>(true);
+  const { nftIds } = props;
+  const [userFolders] = useLocalStorage('user-folders', {});
+  const [userFoldersNFTs, setUserFoldersNFTs] = useLocalStorage('user-folders-nfts', {});
+  const { data: fingerprint } = useGetLoggedInFingerprintQuery();
+  const [, setSelectedNFTIds] = useLocalStorage('gallery-selected-nfts', []);
```
Comment (seeden) [inline packages/gui/src/components/nfts/NFTSaveToFilterDialog.tsx:18]: we are not using gallery-selected-nfts anymore
use useFilteredNFTs instead

File: packages/gui/src/components/nfts/NFTSaveToFilterDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+                <ListItemIcon>
+                  <FolderIcon />
+                </ListItemIcon>
+                {folderName}
+              </MenuItem>
+            ))}
+        </DropdownActions>
+      </DialogContent>
+      <DialogActions>
+        <div style={{ padding: '0 10px 10px 0' }}>
```
Comment (seeden) [inline packages/gui/src/components/nfts/NFTSaveToFilterDialog.tsx:57]: Never use style directly
Never use px directly
Use <Box sx={{ paddingVertical 1 }}

File: packages/gui/src/components/nfts/gallery/NFTGallery.tsx [hunk 9]
```diff
@@ -148,6 +153,44 @@ export default function NFTGallery() {
...
+      lastSelectedUserFolder.current &&
+      userFolder !== lastSelectedUserFolder.current &&
+      nftsSorted.current?.destroy &&
+      isSortableActive.current
+    ) {
+      isSortableActive.current = false;
+      nftsSorted.current?.destroy();
+    }
+    lastSelectedUserFolder.current = userFolder;
+    const sortableParent = document.querySelector('[data-test-id=virtuoso-item-list]');
```
Comment (seeden) [inline packages/gui/src/components/nfts/gallery/NFTGallery.tsx:175]: do not use document.anything directly
use react standard useRef, ref for this everywhere

File: packages/gui/src/components/nfts/gallery/NFTGallery.tsx [hunk 10]
```diff
@@ -262,6 +305,31 @@ export default function NFTGallery() {
   }
   */
 
+  function onSelectUserFolder() {
+    handleSetWalletId(undefined);
+  }
+
+  const sortedNFTs = React.useCallback(() => {
```
Comment (seeden) [inline packages/gui/src/components/nfts/gallery/NFTGallery.tsx:312]: why not to use useMemo instead?

File: packages/gui/src/components/nfts/gallery/NFTGallery.tsx [hunk 11]
```diff
@@ -462,6 +530,16 @@ export default function NFTGallery() {
           </Flex>
         </Flex>
       }
+      sidebar={
+        <NFTProfileDropdown
+          onChange={(id: number | undefined) => {
```
Comment (seeden) [inline packages/gui/src/components/nfts/gallery/NFTGallery.tsx:535]: use just {handleSetWalletId} and check values inside function

File: packages/gui/src/components/nfts/provider/NFTProvider.tsx [hunk 12]
```diff
@@ -528,6 +534,7 @@ export default function NFTProvider(props: NFTProviderProps) {
       }
 
       try {
+        await fetchNFTsCount(undefined);
```
Comment (seeden) [inline packages/gui/src/components/nfts/provider/NFTProvider.tsx:537]: why did you add this here? we are reading count on different place

File: packages/icons/src/NFTs.tsx [hunk 13]
```diff
@@ -4,7 +4,15 @@ import React from 'react';
 import NFTsIcon from './images/NFTs.svg';
 import NFTsSmallIcon from './images/NFTsSmall.svg';
 import CopyIcon from './images/copy.svg';
+import FilterIcon from './images/filter.svg';
```
Comment (seeden) [inline packages/icons/src/NFTs.tsx:7]: Can we use icons from material-ui instead?
Did you check if you can change color of icons properly?

File: packages/icons/src/NFTs.tsx [hunk 14]
```diff
@@ -22,6 +30,67 @@ function CopyIconWithoutFill() {
...
+export function Filter(props: SvgIconProps) {
+  return <SvgIcon component={FilterIconWithoutFill} viewBox="0 0 22 22" {...props} />;
+}
+
+function FolderIconWithoutFill() {
+  return <FolderIcon fill="none" />;
+}
+
+export function Folder(props: SvgIconProps) {
+  return <SvgIcon component={FolderIconWithoutFill} viewBox="0 0 22 22" {...props} />;
```
Comment (seeden) [inline packages/icons/src/NFTs.tsx:51]: you can use property inheritViewBox instead of viewBox

**Review by seeden — CHANGES_REQUESTED** [review]
Please read my comments and update your PR

File: packages/gui/src/components/nfts/NFTCard.tsx (hunk 3 above)
Comment (rifeljm) [reply packages/gui/src/components/nfts/NFTCard.tsx:145]: I don't know how to do it with createPortal :(

Comment (rifeljm) [reply packages/gui/src/components/nfts/NFTCard.tsx:145]: I don't know how to do it with createPortal :(

Comment (rifeljm) [reply packages/gui/src/components/nfts/NFTCard.tsx:145]:
```
  function renderPortal() {
    return createPortal(
      <Box
        id="draggable-card"
        sx={{
          position: 'absolute',
          zIndex: '1000',
          width: `${widthBeforeDrag.current}px`,
          height: `${heightBeforeDrag.current}px`,
          transform: 'scale(0.5)',
          opacity: isHidden ? 1 : 0.6,
        }}
      >
        {renderCard()}
      </Box>,
      document.getElementById('portal') as HTMLElement
    );
  }
  ```

Comment (rifeljm) [reply packages/gui/src/components/nfts/NFTCard.tsx:145]:
```
  function renderPortal() {
    return createPortal(
      <Box
        id="draggable-card"
        sx={{
          position: 'absolute',
          zIndex: '1000',
          width: `${widthBeforeDrag.current}px`,
          height: `${heightBeforeDrag.current}px`,
          transform: 'scale(0.5)',
          opacity: isHidden ? 1 : 0.6,
        }}
      >
        {renderCard()}
      </Box>,
      document.getElementById('portal') as HTMLElement
    );
  }
  ```
